cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesClassloaderUtil.java

```diff
@@ -0,0 +1,351 @@
+     * @param contextConfigurationLocation
+     *            URL to check for classes
+     * @return List of classes to load utilities from
+     */
+    private static List<Class<?>> resolveFromJar(List<Class<?>> result, ClassLoader inputClassLoader,
+        URL contextConfigurationLocation) {
+        LOG.debug("Processing configuration archive {}", contextConfigurationLocation);
+        LOG.info("Searching for classes in configuration archive...");
+
+        List<String> foundClasses = new LinkedList<>();
+        try {
+            // Get the URI of the jar from the URL of the contained context.xml
+            URI jarUri = URI.create(contextConfigurationLocation.toString().split("!")[0]);
+
+            // Make sure to create file system for jar file
+            FileSystem jarfs = FileSystemUtil.getOrCreateFileSystem(jarUri);
+
+            foundClasses = walkJarFile(jarUri, jarfs);
+        } catch (IOException e) {
+            LOG.error("Could not read templates jar file", e);
+        }
+        if (foundClasses.size() > 0) {
+            for (String className : foundClasses) {
+                try {
+                    result.add(inputClassLoader.loadClass(className));
+                } catch (ClassNotFoundException e) {
+                    LOG.warn("Could not load " + className + " from classpath");
+                    LOG.debug("Class was not found", e);
+                }
+            }
+        } else {
+            LOG.info("Could not find any compiled classes to be loaded as util classes in jar file.");
+        }
+        return result;
+    }
+
+    /**
+     * Walks the template folder in search of utility classes
+     *
+     * @param templateRoot
+     *            Path to template folder
+     * @return List<Path> of paths containing a class file
+     * @throws IOException
+     *             if file could not be visited
+     */
+    private static List<Path> walkTemplateFolder(Path templateRoot) throws IOException {
+        final List<Path> foundPaths = new LinkedList<>();
+        Files.walkFileTree(templateRoot, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.toString().endsWith(".class")) {
+                    foundPaths.add(file);
+                    LOG.debug("    * Found class file {}", file);
+                }
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                // Log errors but do not throw an exception
+                LOG.warn("An IOException occurred while reading file on path {} with message: {}", file,
+                    exc.getMessage());
+                LOG.debug("An IOException occurred while reading file on path {} with message: {}", file,
+                    LOG.isDebugEnabled() ? exc : null);
```

Comment:
You can put this last parameter simply in the warn log before

Reply:
adjusted.